src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -401,3 +401,46 @@ async def outbound_call(self, inputs_info: tuple[Job, ...]):
                     if not fut.done():
                         fut.cancel()
             self._sema.release()
+
+
+class NoOpDispatcher:
+    """
+    NoOpDispatcher is a dispatcher that dispatches requests to the user defined
+    callback function as soon as the request arrives. It does not do any batching
+    or optimization.
+    """
+
+    def __init__(self):
+        pass
+
+    def shutdown(self):
+        pass
+
+    async def inbound_call(self, data: t.Any):
+        return self.callback((data,))
+
+    async def outbound_call(self):
+        pass
+
+    async def controller(self):
+        pass
+
+    def __call__(
+        self,
+        callback: t.Callable[
+            [t.Sequence[T_IN]], t.Coroutine[None, None, t.Sequence[T_OUT]]
+        ],
+    ) -> t.Callable[[T_IN], t.Coroutine[None, None, T_OUT]]:
+        self.callback = callback
+
+        @functools.wraps(callback)
+        async def _func(data: t.Any) -> t.Any:
+            try:
+                r = await self.inbound_call(data)
+            except asyncio.CancelledError:
+                return None
+            if isinstance(r, Exception):
+                raise r
+            return r
```

Comment:
```suggestion
            try:
                return await self.inbound_call(data)
            except Exception as err:
            	logger.error("Exception caught while sending request:\n%s", err, exc_info=err)
            	raise err from None
```

Reply:
Hm, did `inbound_call` return an exception before? What line 442 for originally?

Reply:
I was copying from https://github.com/bentoml/BentoML/blob/main/src/bentoml/_internal/marshal/dispatcher.py#L181